core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -307,6 +307,13 @@ private void beginSnapshot(ClusterState clusterState, final SnapshotsInProgress.
                 metaData = builder.build();
             }
 
+            // remove allocation ids from metadata
+            MetaData.Builder builder = MetaData.builder(metaData);
+            for (ObjectObjectCursor<String, IndexMetaData> index : metaData.getIndices()) {
+                builder.put(IndexMetaData.builder(index.value).removeAllPrimaryCandidateAllocationIds().build(), false);
```

Comment:
this is a good one. Do you know what shard state we generate when we are restoring?